### PR TITLE
feature(atomic long): Implement a distributed atomic long

### PR DESCRIPTION
**Pull request by phoebesimon**
*How*
orca.js: Added functions to get/increment/decrement a long
test-orca.js: Tests for the above functions
README.md: Small step missing

*Why*
Useful for many coordination tasks

### CODE REVIEW

File: test-orca.js [hunk 1]
```diff
@@ -308,6 +308,193 @@ test('test monitor (add 3, remove 1)', function(t) {
...
+test('get long', function(t) {
+  var path = '/bar-' + randomstring.generate(8) ,
+      cxn;
+
+  cxn = zkorca.getCxn(defaultOptions());
+
+  async.auto({
+    get: function(callback) {
+      cxn.get(path, function(err, data) {
+        t.ok(err, 'Cannot get a value before it has been instantiated');
```
Comment (phoebesimon) [inline test-orca.js]: I wasn't exactly clear on what should happen in this case: if it has never been instantiated, should we error or assume they want to instantiate it?

File: orca.js [hunk 2]
```diff
@@ -380,6 +380,204 @@ ZkOrca.prototype._doubleBarrierLeave = function(path, callback) {
...
+  var self = this,
+      dataFilename, lockFilename;
+
+  if (name[name.length - 1] === '/') {
+    callback(new Error('Path name may not end in \'/\''));
+    return;
+  }
+
+  dataFilename = name + '/data';
+  lockFilename = name + '/lock';
```
Comment (rphillips) [inline orca.js]: ```dataFilename = /somePath/ele-21412/data```

The name ele-21412 is non-ephemeral, so the directory would linger around. A better pathname would be

```path.join('/', self._name, 'locks', name);```
*where self._name is what purpose it is.*

so you get a path '/some_usage/locks/...' and /some_usage/data/...'

Comment (phoebesimon) [inline orca.js]: @rphillips I see what you're getting at, fixed in 90a02a8

File: test-orca.js (hunk 1 above)
Comment (rphillips) [inline test-orca.js]: perhaps passing a default value if it isn't instantiated, and erroring out if a default is not present.

Comment (phoebesimon) [inline test-orca.js]: @rphillips do you mean making a optional parameter? so something like 
```
ZkOrca.get(path, default, callback) {
  if (!callback && typeof default === 'function') {
    callback = default;
    default = null;
  }
  
  if (default) {
    //attempt to get, setting data to default if not present
  } else {
    //attempt to get, error if not present
  }
}
```

Comment (phoebesimon) [inline test-orca.js]: @rphillips let me know what you think 9fdf291

**Comment by rphillips** [comment]
very nice. +1

**Pull request merged**